drv/sidecar-front-io-api/build.rs

```diff
@@ -0,0 +1,166 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use serde::Deserialize;
+use std::fmt::Write;
+use std::{env, fs, path::PathBuf};
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    build_util::expose_target_board();
+
+    let out_dir = &PathBuf::from(env::var_os("OUT_DIR").unwrap());
+
+    fs::write(out_dir.join("sidecar_qsfp_x32_controller.rs"), regs()?)?;
+
+    let ecp5_bitstream_name = match env::var("HUBRIS_BOARD")?.as_str() {
+        "sidecar-1" => "sidecar_qsfp_x32_controller.bit",
+        _ => {
+            println!("No FPGA image for target board");
+            std::process::exit(1)
+        }
+    };
+    let fpga_bitstream = fs::read(ecp5_bitstream_name)?;
+    let compressed_fpga_bitstream = compress(&fpga_bitstream);
+
+    fs::write(out_dir.join("ecp5.bin.rle"), &compressed_fpga_bitstream)?;
+
+    // Make sure the app image is rebuilt if the bitstream file for this target
+    // changes.
+    println!("cargo:rerun-if-changed={}", ecp5_bitstream_name);
+
+    Ok(())
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(tag = "type", rename_all = "lowercase")]
+enum Node {
```

Comment:
This code is now duplicated 3x (`gimlet-seq-server/build.rs`, `sidecar-mainboard-controller-api`, and `sidecar-front-io-api`).  It's probably time to make a separate crate (`build/fpga-regmap`?) and call into it!

Reply:
Fair, I'll do that.